### PR TITLE
Refactor facilitator to prepare for m1

### LINKED ISSUE #278: Directory restructuring
- Restructure repository to support different personas 
- Move everything for m0-facilitator
      - For `src`, `test` and `test_integration`

**Please note: using `git mv` command to move the files**

Proposed structure:
```
    facilitator
        |
        |-----src
        |      |----bin
        |      |
        |      |----common (code shared by both gen0 and gen1)
        |      |      |  
        |      |      |-- subscriber
        |      |      .      .
        |      |      .      .
        |      |        
        |      |----mo-faciltiator
        |      |      |-- models
        |      |      |-- services
        |      |      |-- handlers
        |      |      .      .
        |      |      .      .        
        |      |        
        |      |----m1-faciltiator
        |             |-- models
        |             |-- services
        |             |-- handlers
        |             .      .
        |             .      .        
        |      
        |-----test
        |      |----m0-facilitator
        |      |----m1-facilitator
        |      .     .
        |      
        |-----test_integration
        |      |----m0-facilitator
        |      |----m1-facilitator
        |      .     .
        |-----testnet
        |        |-gen0_manifest
        |           |- m1-facilitator.yaml
        |
        |-----package.json
        |-----index.ts
        .      .
        .      .
    ```

### PR DESCRIPTION
**Pull request by 0xsarvesh**
fixes #278 

1. Repo restructuring

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,20 +14,20 @@
     "package": "./package.sh",
     "prepack": "npm run package",
     "test:unit": "mocha --require ts-node/register --require source-map-support/register --recursive test/**/**/**/*.test.ts test/**/**/*.test.ts test/**/*.test.ts test/*.test.ts",
```
Comment (abhayks1) [inline package.json]: with new directory structure test needs to run for 1 more nested `*.test.ts` files.
e.g for services tests 
`test/**/**/**/**/*.test.ts`  it needs to be updated

File: test/m0-facilitator/package.sh [hunk 2]
```diff
@@ -40,6 +40,6 @@ echo "Accessing as binary."
 ./node_modules/.bin/facilitator help || exit 1
 
 echo "Cleaning up generated files."
-clean
+#clean
```
Comment (abhayks1) [inline test/m0-facilitator/package.sh]: why this 🤔 ?

**Review by abhayks1 — CHANGES_REQUESTED** [review]
There are inline comments 👍 

- We can also update directory structure of demo. demo/bin/m0_facilitator
m1_facilitator demo scripts will be separate.

- Also please enable test suite `npm run test:package` in travis.yml.
I had to comment that in travis since u were already looking into that issue to avoid duplicate effort. I made comment here referring your ticket: https://github.com/mosaicdao/facilitator/pull/309#issuecomment-587469416

File: .travis.yml [hunk 3]
```diff
@@ -11,6 +11,7 @@ env:
     - DOCKER_COMPOSE_VERSION=1.23.2
   jobs:
     - TEST_SUITE="test:unit"
+    - TEST_SUITE="test:package"
```
Comment (abhayks1) [inline .travis.yml:14]: 👍 💯

File: test/m0-facilitator/package.sh (hunk 2 above)
Comment (0xsarvesh) [reply test/m0-facilitator/package.sh]: Corrected.

File: package.json (hunk 1 above)
Comment (0xsarvesh) [reply package.json]: Added

**Review by abhayks1 — CHANGES_REQUESTED** [review]
Looks good 💯 
1 final comment to also restructure testdata as m0, m1 testdata will be completely different. e.g. mosaic config and gateway config doesn't exist in m1 facilitator.

It's good to merge afterwards.

testdata/m0_facilitator

**Review by abhayks1 — APPROVED** [review]

**Pull request merged**